Service/src/main/java/org/apidb/apicommon/service/services/jbrowse/JBrowseUserDatasetsService.java

```diff
@@ -1,161 +1,206 @@
 package org.apidb.apicommon.service.services.jbrowse;
 
 import javax.ws.rs.GET;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.NotFoundException;
 import javax.ws.rs.Path;
 import javax.ws.rs.PathParam;
 import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
 import javax.ws.rs.core.MediaType;
 
 import org.apache.log4j.Logger;
+import org.apidb.apicommon.service.services.jbrowse.model.JBrowseDatasetResponse;
+import org.apidb.apicommon.service.services.jbrowse.model.JBrowseTrack;
+import org.apidb.apicommon.service.services.jbrowse.model.VDIDatasetReference;
+import org.apidb.apicommon.service.services.jbrowse.model.VDIDatasetType;
+import org.gusdb.fgputil.db.runner.SQLRunner;
 import org.gusdb.fgputil.events.Events;
 import org.gusdb.wdk.errors.ErrorContext.ErrorLocation;
 import org.gusdb.wdk.errors.ServerErrorBundle;
 import org.gusdb.wdk.events.ErrorEvent;
 import org.gusdb.wdk.model.WdkModelException;
 import javax.ws.rs.ForbiddenException;
-import org.gusdb.wdk.model.user.dataset.UserDataset;
-import org.gusdb.wdk.model.user.dataset.UserDatasetDependency;
-import org.gusdb.wdk.model.user.dataset.UserDatasetFile;
-import org.gusdb.wdk.model.user.dataset.UserDatasetInfo;
-import org.gusdb.wdk.model.user.dataset.UserDatasetMeta;
-import org.gusdb.wdk.model.user.dataset.UserDatasetSession;
-import org.gusdb.wdk.model.user.dataset.UserDatasetType;
-import org.gusdb.wdk.service.formatter.UserDatasetsFormatter;
-import org.gusdb.wdk.service.service.user.UserDatasetService;
+import javax.ws.rs.core.Response;
+
 import org.gusdb.wdk.service.service.user.UserService;
-import org.json.JSONArray;
-import org.json.JSONObject;
+
+import java.io.File;
+import java.nio.file.Paths;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static org.gusdb.wdk.service.FileRanges.getFileChunkResponse;
+import static org.gusdb.wdk.service.FileRanges.parseRangeHeaderValue;
 
 public class JBrowseUserDatasetsService extends UserService {
 
   private static final Logger LOG = Logger.getLogger(JBrowseUserDatasetsService.class);
+  private static final String VDI_DATASET_DIR_KEY = "VDI_DATASETS_DIRECTORY";
+  private static final String VDI_CONTROL_SCHEMA_KEY ="VDI_CONTROL_SCHEMA";
+  public static final String BIGWIG_SUBCATEGORY = "Bigwig Files From User";
```

Comment:
Should these be in the enum below?

Reply:
Yep, fixed and added a method to translate the db value to the enum.